### PR TITLE
Support formatter/parser plugins

### PR DESCRIPTION
**Pull request by cosmo0920**
Use new Plugin.new_formatter API.

### CODE REVIEW

File: lib/fluent/plugin/in_named_pipe.rb [hunk 1]
```diff
@@ -25,8 +25,8 @@ def configure(conf)
         raise ConfigError, "#{e.class}: #{e.message}"
       end
 
-      config['format'] = @format
-      configure_parser(conf)
+      @formatter = Plugin.new_formatter(@format)
+      @formatter.configure(conf)
```
Comment (sonots) [inline lib/fluent/plugin/in_named_pipe.rb]: not parser?

Comment (cosmo0920) [inline lib/fluent/plugin/in_named_pipe.rb]: Oops, it's my mistake.

**Comment by cosmo0920** [comment]
Should I squash commits?

**Comment by sonots** [comment]
Can you fix to use parser, too along the way?

**Comment by cosmo0920** [comment]
Hmm, you mean that "use Plugin.new_parser instead of using TextParser.new"?
If so, I'll try it!

File: lib/fluent/plugin/in_named_pipe.rb [hunk 2]
```diff
@@ -25,12 +25,7 @@ def configure(conf)
         raise ConfigError, "#{e.class}: #{e.message}"
       end
 
-      config['format'] = @format
-      configure_parser(conf)
-    end
-
-    def configure_parser(conf)
-      @parser = TextParser.new
+      @parser = Plugin.new_parser(conf['format'])
```
Comment (sonots) [inline lib/fluent/plugin/in_named_pipe.rb]: prefer to use `@format` for symmetricity with formatter.

**Comment by cosmo0920** [comment]
Thanks for the suggestions and merging!

**Comment by sonots** [comment]
released 0.1.1!

**Comment by cosmo0920** [comment]
Yay! :clap:

**Pull request merged**